File: Leetcode/stack_queue/71_SimplifyPath/Solution.java
class Solution {
    public String simplifyPath(String path) {
        String[] paths = path.split("/");
        Stack<String> stack = new Stack<>();
        for(String s: paths){
            if(s.equals("..")){
                if(!stack.empty() )
                    stack.pop();
            }
            else if(s.equals(".") || s.length()==0){
                continue;
            }else{
                stack.push(s);
            }
        }
        String res = "";
        while(!stack.empty()) res = "/"+stack.pop()+res;
        return (res.length()==0)?"/":res;
    }
}

File: Leetcode/stack_queue/71_SimplifyPath/Solution.py
class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        stack = []
        paths = list(filter(lambda s: s!='', path.split('/')))
        for token in paths:
            if token=='..':
                if stack:
                    stack.pop()
            elif token == '.':
                continue
            else:
                stack.append(token)
        return "/"+"/".join(stack)
